# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: layout/base/MotionPathUtils.cpp
# Commit: ec44194caefb
# Full Hash: ec44194caefb41931247d18ad886326cf6d89a93
# Author: Boris Chiou <boris.chiou@gmail.com>
# Date: 2024-03-05 09:48:50
# Description:
#   Bug 1870200 - Break the cyclic dependency if any of the desendants uses non-scaling-stroke. r=emilio
#   
#   While we are computing the transform-box:stroke-box, for CSS Transforms or
#   Motion path Transforms, we have to avoid the cyclic dependency not only
#   for the SVG geometry frame itself, but also for the desendants of
# ==============================================================================

diff -r 5c84d062c789 -r ec44194caefb layout/base/MotionPathUtils.cpp
--- a/layout/base/MotionPathUtils.cpp	Mon Mar 04 20:33:13 2024 +0000
+++ b/layout/base/MotionPathUtils.cpp	Mon Mar 04 20:46:05 2024 +0000
@@ -111,9 +111,11 @@
   const auto size = CSSSize::FromAppUnits(
       (aFrame->HasAnyStateBits(NS_FRAME_SVG_LAYOUT)
            ? nsLayoutUtils::ComputeSVGReferenceRect(
-                 aFrame, aFrame->StyleSVGReset()->HasNonScalingStroke()
-                             ? StyleGeometryBox::FillBox
-                             : StyleGeometryBox::StrokeBox)
+                 aFrame,
+                 aFrame->StyleSVGReset()->HasNonScalingStroke()
+                     ? StyleGeometryBox::FillBox
+                     : StyleGeometryBox::StrokeBox,
+                 nsLayoutUtils::MayHaveNonScalingStrokeCyclicDependency::Yes)
            : nsLayoutUtils::ComputeHTMLReferenceRect(
                  aFrame, StyleGeometryBox::BorderBox))
           .Size());